app/lib/submission_builder/ty2024/states/nj/documents/nj1040.rb

```diff
@@ -137,6 +137,18 @@ def document
                     xml.TotalIncome calculated_fields.fetch(:NJ1040_LINE_27)
                   end
 
+                  if calculated_fields.fetch(:NJ1040_LINE_28A).positive? && Flipper.enabled?(:show_retirement_ui)
+                    xml.PensionExclusion calculated_fields.fetch(:NJ1040_LINE_28A)
+                  end
+
+                  if calculated_fields.fetch(:NJ1040_LINE_28B).positive? && Flipper.enabled?(:show_retirement_ui)
+                    xml.OtherRetireIncomeExclus calculated_fields.fetch(:NJ1040_LINE_28B)
+                  end
+
+                  if calculated_fields.fetch(:NJ1040_LINE_28C).positive? && Flipper.enabled?(:show_retirement_ui)
+                    xml.TotalExclusionAmount calculated_fields.fetch(:NJ1040_LINE_28C)
+                  end
+
```

Comment:
[sand] Now that we're in the more-than-two case, I'd propose a refactoring here where we group all these fields into a single `if Flipper enabled` check (including lines 128-132) rather than redoing this conditional on each individual line